src/python/pants/backend/codegen/protobuf/python/rules.py

```diff
@@ -111,8 +126,15 @@ async def generate_python_from_protobuf(
                 bin_names=[protoc_gen_mypy_script],
             ),
         )
+        protoc_argv.extend(
+            [
+                f"--plugin=protoc-gen-mypy={mypy_pex.bin[protoc_gen_mypy_script].argv0}",
+                "--mypy_out",
+                output_dir,
+            ]
+        )
 
-        if request.protocol_target.get(ProtobufGrpcToggleField).value:
+        if grpc_enabled and python_protobuf_subsystem.grpcio_plugin:
```

Comment:
Does grpclib generate annotated code, or type stubs? Can its generated code be made to typecheck with mypy?

Reply:
Yeah, it's typed python. If you are curious, it looks like this: https://gist.github.com/isra17/12f3a6c7cd2df3635ff36c3a72309f49

Reply:
Nice!